toucan_connectors/google_big_query/google_big_query_connector.py

```diff
@@ -112,6 +120,35 @@ def _define_query_param(name: str, value: Any) -> BigQueryParam:
 class GoogleBigQueryConnector(ToucanConnector, DiscoverableConnector):
     data_source_model: GoogleBigQueryDataSource
 
+    @validator('credentials', pre=True, always=True)
+    def validate_google_credentials(cls, value):
+        # list of fields that can be missing or None
+        optional_fields = [
+            'private_key_id',
+            'private_key',
+            'client_email',
+            'client_id',
+            'auth_uri',
+            'auth_provider_x509_cert_url',
+            'client_x509_cert_url',
+        ]
+
+        # Iterate over optional fields and set them to None if missing
+        for field_name in optional_fields:
+            if not hasattr(value, field_name) and not value.get(field_name):
+                # for all links
+                if field_name in [
+                    'client_x509_cert_url',
+                    'auth_provider_x509_cert_url',
+                    'auth_uri',
+                ]:
+                    value[field_name] = 'https://valid-scheme.com'
+                else:
+                    # for other values
+                    value[field_name] = '__not_set__'
```

Comment:
is there no other way of doing that? seems a bit odd

Reply:
And you're right, after discussing with Thomas, we all agreed we should have a field "project-id" for the jwt,  so I created a second model called JWTCredentials and remove all those "WEIRDS" validators we were having

Reply:
Please see below a demo : 
![test](https://github.com/ToucanToco/toucan-connectors/assets/22576758/85bcdc1c-3cbf-408b-bbae-7a0be3cd867b)
let me know what you think

Reply:
Either we check for the "GoogleCredentials authentification" either we check for the "JWTCredentials authentication" and then fill all those required values.